File: app/src/main/java/com/amir/alzheimer/infrastructure/gallery/PeopleAdapter.java
package com.amir.alzheimer.infrastructure.gallery;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.amir.alzheimer.R;

import java.util.ArrayList;

public class PeopleAdapter extends BaseAdapter {
    private final ArrayList<String> peoples;
    private final LayoutInflater inflater;

    public PeopleAdapter(Context context, ArrayList<String> peoples) {
        this.inflater = LayoutInflater.from(context);
        this.peoples = peoples;

        peoples.add("Amir");
        peoples.add("Mehdi");
        peoples.add("Mohsen");
        peoples.add("Tannaz");
        peoples.add("Zeynab");

    }


    @Override
    public int getCount() {
        return peoples.size();
    }

    @Override
    public String getItem(int position) {
        return peoples.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        PeopleViewHolder viewHolder;
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.people_item, parent, false);
            viewHolder = new PeopleViewHolder();
            viewHolder.textView = (TextView) convertView.findViewById(R.id.people_item_text_view);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (PeopleViewHolder) convertView.getTag();
        }

        viewHolder.textView.setText(peoples.get(position));

        return convertView;
    }


    class PeopleViewHolder {
        TextView textView;
    }
}

File: app/src/main/java/com/amir/alzheimer/infrastructure/gallery/PeopleAdapter.kt
package com.amir.alzheimer.infrastructure.gallery

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.amir.alzheimer.R
import com.amir.alzheimer.infrastructure.database.AlzhimerDatabase
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.rxkotlin.addTo
import io.reactivex.schedulers.Schedulers
import io.reactivex.disposables.CompositeDisposable



class PeopleAdapter(context: Context?) : BaseAdapter() {
    private val inflater: LayoutInflater = LayoutInflater.from(context)
    private var peoples: List<String> = listOf()
    var compositeDisposable = CompositeDisposable()

    init {
        AlzhimerDatabase.getAlzhimerDao().allRelatives()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { relatives ->
                    peoples = relatives.map { it.title }
                    Log.wtf("Aaaa", "aaaaa")
                    notifyDataSetChanged()
                }.addTo(compositeDisposable)

    }

    override fun getCount(): Int {
        return peoples.size
    }

    override fun getItem(position: Int): String {
        return peoples[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val viewHolder: PeopleViewHolder
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.people_item, parent, false)
            viewHolder = PeopleViewHolder()
            viewHolder.textView = convertView!!.findViewById(R.id.people_item_text_view)
            convertView.tag = viewHolder
        } else {
            viewHolder = convertView.tag as PeopleViewHolder
        }

        viewHolder.textView!!.text = peoples[position]

        return convertView
    }


    internal inner class PeopleViewHolder {
        var textView: TextView? = null
    }
}
